content/en/docs/concepts/services-networking/topology-aware-hints.md

```diff
@@ -0,0 +1,127 @@
+---
+reviewers:
+- robscott
+title: Topology Aware Hints
+feature:
+  title: Topology Aware Hints
+  description: >
+    Enables topology aware routing with topology hints on EndpointSlices.
+
+content_type: concept
+weight: 10
+---
+
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.21" state="alpha" >}}
+
+_Topology Aware Hints_ enable topology aware routing with topology hints
+included in EndpointSlices. This approach tries to keep traffic close to where
+it originated from. This can result in decreased costs and improved performance.
+
+<!-- body -->
+
+## Introduction
+
+Kubernetes clusters are increasingly deployed in multi-zone environments.
+_Topology Aware Hints_ adds new functionality that can help keep traffic within
+the zone it originated from. This concept is commonly referred to as "Topology
+Aware Routing". The "Hints" here refer to a key implementation detail. The
+EndpointSlice controller adds hints to EndpointSlices to allocate each endpoint
+to a zone. Those hints are then consumed by components like kube-proxy as they
+configure how requests are routed.
+
+## Using Topology Aware Hints
+
+This feature can be enabled for individual Services by setting the
+`service.kubernetes.io/topology-aware-hints` annotation to `auto`. This tells
+the EndpointSlice controller to set topology hints if it is deemed safe.
+Importantly, this does not guarantee that hints will always be set.
+
+## How it Works
```

Comment:
Optional improvement: I think it'd really help to have diagram showing the difference in traffic (or similar) for a Service with or without `service.kubernetes.io/topology-aware-hints`. After reading this page once I don't think I could draw that diagram but if it was already there I think my understanding would be a lot more thorough.

BTW, we can use Mermaid for diagrams so you don't need to use graphics software.

Reply:
Agree, this is a good idea, will try to set aside some time for this. Do you have any examples of diagrams that have been done well in k8s docs? Is Mermaid optional?

Reply:
https://kubernetes.io/docs/concepts/services-networking/ingress/ is my usual example of diagrams in Mermaid.

https://kubernetes.io/docs/concepts/overview/components/ shows our “house style”; it uses icons from https://github.com/kubernetes/community/tree/master/icons

Reply:
Actually I notice there's no icon for EndpointSlice yet.

Reply:
It's OK to add the diagram later, eg when this is scheduled to go beta.

Reply:
Been overloaded this week so I don't think I'll get time to add this in for this release, but will keep it in mind for beta, agree that it would be helpful.